engine/src/main/java/org/terasology/rendering/nui/layouts/FlowLayout.java

```diff
@@ -51,16 +58,17 @@ public void removeAllWidgets() {
     public void onDraw(Canvas canvas) {
         int filledWidth = 0;
         int filledHeight = 0;
+        int widthOffset = 0;
         int heightOffset = 0;
         for (UIWidget widget : contents) {
             Vector2i size = canvas.calculatePreferredSize(widget);
-            if (filledWidth != 0 && filledWidth + size.x  > canvas.size().x) {
+            if (filledWidth != 0 && filledWidth + size.x > canvas.size().x) {
```

Comment:
I think this check also needs to take the spacing into account. If we add both the spacing and the size of the next element to the currently filled width, do we exceed the canvas size?

Reply:
Considering we add the spacing right afterwards, yes that probably should include the spacing as well. Generally though, wouldn't we want for the item to be added if it fits to the row, no matter the spacing we set in between the other items? 

So what my understanding of the code is that it adds the item and the spacing (to its right) in one loop, and then go to the next item. Is that how it works? Because I still feel like there is a detail within the two methods that I still haven't understood yet.

Reply:
Yeah, you're probably right - that's also where "remove from last element" comes from 🤔  so this check is indeed correct, because the "previous" gap is already contained in `filledWidth`... 👍